hierarchicalforecast/utils.py

```diff
@@ -190,14 +194,40 @@ def aggregate(
             "Don't set this argument to suppress this warning.",
             category=DeprecationWarning,
         )
+    
+    # Define acceptable aggregation functions
+    acceptable_aggregations = {
+        'sum', 'mean', 'median', 'min', 'max', 'count', 'std', 'var', 'first', 'last'
+    }    
+    # Check if exog_vars are present in df
+    if exog_vars is not None:
+        missing_vars = [var for var in exog_vars.keys() if var not in df.columns]
+        if missing_vars:
+            raise ValueError(f"The following exogenous variables are not present in the DataFrame: {', '.join(missing_vars)}")
+      
+        # Check if the aggregation functions are acceptable
+        invalid_aggregations = [agg for agg in exog_vars.values() if agg not in acceptable_aggregations]
+        if invalid_aggregations:
+            raise ValueError(f"The following aggregation functions are not acceptable: {', '.join(invalid_aggregations)}")       
             
     # compute aggregations and tags
     spec = sorted(spec, key=len)
     bottom = spec[-1]
     aggs = []
     tags = {}
+    # Prepare the aggregation dictionary
+    agg_dict = {
+        "y": ("y", "sum")
+    }
+
+    # Add exog_vars to the aggregation dictionary if it is not None
+    if exog_vars is not None:
+        agg_dict.update({key: (key, exog_vars[key]) for key in exog_vars.keys()})
```

Comment:
Could you please give an example usage of `exog_vars` in this context?

I have not used pandas much lately, but I think that given your type signature of `Dict[str, str]`, you intend to have `exog_vars = {"col_a": "sum", "col_b": "sum"}`. However, this does not support multiple functions to aggregate a particular column as you are going down the named aggregation route.

A way around this will be either `exog_vars = {"col_a": ("sum", "mean")}` which will create a `MultiIndex`, or alternatively something like `exog_vars = {"col_a_sum": ("col_a", "sum"), "col_a_mean": ("col_a", "mean")}`. Either way, the distinction between the output column name for the aggregation and the column name to be aggregated will need to be made when inserting into `agg_dict` to avoid overwriting anything in this case.